cuda/solver/cb_gmres_kernels.cu

```diff
@@ -479,6 +482,9 @@ void step_2(std::shared_ptr<const CudaExecutor> exec,
             matrix::Dense<ValueType>* before_preconditioner,
             const Array<size_type>* final_iter_nums)
 {
+    if (before_preconditioner->get_size()[1] == 0) {
```

Comment:
Just a comment, not holding anything up, but would it make sense to check for that (i.e. `num_rhs > 0`) right at the beginning of the solver application? (That also holds for the other solvers, and it should reduce the repetition across solver steps and devices)

Reply:
Most of this code will change with #861, so I didn't want to spend too much time working on it.